packages/sync-repo-settings/src/sync-repo-settings.ts

```diff
@@ -83,15 +62,39 @@ function handler(app: Application) {
     const name = context.payload.repository.name;
     const repo = `${owner}/${name}`;
 
-    // find the repo record in repos.json
-    const repos = await handler.getRepos();
-    const yoshiRepo = repos.repos.find(x => x.repo === repo);
-    console.log(yoshiRepo?.repo);
-    if (!yoshiRepo) {
+    // Fetch the list of languages used in this repository
+    const langRes = await context.github.repos.listLanguages({
+      owner,
+      repo: name,
+    });
+
+    // Given an object like this:
+    // {
+    //   python: 24
+    //   javascript: 22
+    // }
+    // Sort the keys based on instance count.
+    const languages = Object.entries(langRes.data).sort((a, b) => {
+      return a[1] > b[1] ? -1 : a[1] < b[1] ? 1 : 0;
```

Comment:
Consider:

```suggestion
      return a[1] - b[1];
```

Reply:
This is a kinda weird API where `1` means `a` is greater than `b`, `-1` means `b` is greater than `a`, and `0` means `a` equals `b`.  We are doing it backwards so the highest number comes first.  I don't think a subtraction would work.

Reply:
Ah, I missed it was swapped. `b[1] - a[1]` should work. It can be any value greater or less than 0 -- doesn't need to be 1 or -1 exactly.

| Relation | Result |
| --- | --- |
| a > b | b - a < 0 |
| a = b | b - a = 0 |
| a < b | b - a > 0 |

Reply:
Ah, that makes a lot more sense!  Nice find.  Will include this in a follow PR I'm building now.